esphomeyaml/cookbook/bme280_environment.rst

```diff
@@ -0,0 +1,117 @@
+BME280 Environment
+==================
+
+.. seo::
+    :description: Instructions for setting up BME280 sensors in esphomelib and calculate altitude, absolute humidity, and sea level pressure.
+    :image: bme280.jpg
+    :keywords: BME280
+
+The :doc:`</esphomeyaml/components/sensor/bme280>` is a simple temperature, humidity, and pressure sensor with communication over I2C.
+With some simple math it is possible to either determine the height of the sensor, or the current pressure at sea level.
+This guide can be applied to any sensor measuring temperature and pressure at the same time, like the :doc:`</esphomeyaml/components/sensor/bmp280>`, or :doc:`</esphomeyaml/components/sensor/bme680>`.
+
+.. figure:: images/bme280-header.jpg
+    :align: center
+    :width: 75.0%
+
+The first step is to connect the sensor as described :doc:`here </esphomeyaml/components/switch/bme280>`.
+After validating the sensor is working, we can proceed and add some formulas.
+
+.. code-block:: yaml
+    globals:
+      - id: standard_sea_level_pressure_hpa
+        type: float
+        restore_state: no
+        initial_value: '1013.25'
```

Comment:
Why use a global here? Globals have quite a overhead (well not too big, but certainly more than local variables). I think it would be simpler just to do something like this:

```yaml
- platform: template
  # ...
  lambda: |-
    float STANDARD_SEA_LEVEL_PRESSURE = 1013.25;  // in hPa
    return ((id(bme280_temperature).state + 273.15) / 0.0065) * ((powf(id(standard_sea_level_pressure) / id(bme280_pressure).state), 0.190234) - 1);
```

Reply:
The idea was that this can be replaced by an Mqtt entry fetching the standard sea level pressure real time from either a stationary sensor in your network, or the web (via HA or something). I did not yet wrote that part as it involves a lot more (Mqtt, ha, etc).